lib/auth.js

```diff
@@ -117,7 +120,10 @@ Auth.getAuth = function getAuth(req, res, next) {
  */
 Auth.authenticate = function authenticate(req, res, next) {
     var params = req.body;
-    if (typeof (params.username) === 'undefined') {
+    var username = params.username;
+    USERNAME_LOGIN_ATTEMPTS[username] = req.ufdsSlave.connected && !req.ufdsMaster.connected &&
+        USERNAME_LOGIN_ATTEMPTS[username] || 1;
```

Comment:
I think 124-125 would be clearer as an `if`/`else` block, e.g.

```
if (!req.ufdsSlave.connected && req.ufdsMaster.connected) {
    USERNAME_LOGIN_ATTEMPTS[username] = 1;
}
```

It would also be good to reflect _why_ you are doing this in a comment.